### ISSUE TITLE
Ability to Purchase Additional Space For groups

### ISSUE THREAD

**Issue by win189** [Neos-Metaverse/NeosPublic#3303]
### Is your feature request related to a problem? Please describe.

#1616

### Relevant issues

that issue was per user I would like for a group encouraging Community building

### Describe the solution you'd like

to have the ability

### Describe alternatives you've considered

Beg and bug shifty for more space

**Comment by shadowpanther** [comment]
More general - the ability to convert your personal space into group space, as having over 100G of space while only getting 10G for your group might not be something everyone wants. Maybe have some conversion ratio, like 2G of personal space equals 1G of group space as it has more benefits (multiple user access).

**Comment by Frooxius** [comment]
I think both are things we could do. Buying storage for groups seems sensible, since not everyone would do it on Patreon either and it should give more flexibility. It's something we plan to do regardless.

The Patreon conversion is trickier - we could add system which says how is this distributed among your groups. Perhaps you give them a "weight" and anytime Patreon updates, it calculates the proportion of the space and redistributes by how you want it? Would that make most sense? It's pretty much the ratio expressed in weights.

**Comment by win189** [comment]
Couldn't we do it where its a panel that shows your total space  and you simply put a value in for your group storage and have on display what is left and that is what is your userspace I feel that would make it easier for the end-user and possibly for your coding 

also, it would be cool if members of a group could donate some of their space provided they are patrons of course so that you don't have a person making 100 accounts and donating there 1 GB of free storage

**Comment by Frooxius** [comment]
We'll need new UI for that first. It depends on how soon we want to introduce this.

We could do the weighting through Neos Bot commands for the time being if we want this fast. It's not going to be best UI/UX for it, but it'll be something.

**Comment by shadowpanther** [comment]
Yeah, something along the lines of "Allocate [N Gb/N %] of my personal storage towards G-GroupName shared storage". Probably add a cooldown to this change, so that it couldn't be exploited much. 
This might of course be postponed until the group management interface is done (and done manually by the staff in the meanwhile), or made into a bot command, as I don't expect this allocation to be changed often.
